### PR TITLE
Pull metrics server for registry.k8s.io for k-sigs/cluster-api-provider-azure

### PR DESCRIPTION
**Pull request by marosset**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes/k8s.io/issues/4738

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by bridgetkromhout** [comment]
Possibly related: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3035

File: templates/addons/metrics-server/metrics-server.yaml [hunk 1]
```diff
@@ -138,7 +138,7 @@ spec:
         - --kubelet-use-node-status-port
         - --metric-resolution=15s
         - --kubelet-insecure-tls
-        image: k8s.gcr.io/metrics-server/metrics-server:v0.6.2
+        image: registry.k8s.io/metrics-server/metrics-server:v0.6.2
```
[the remark this answers is not in the document]
Comment (marosset) [reply templates/addons/metrics-server/metrics-server.yaml:141]: I liked an issue and PR from the metrics-server repo to 3035.
Once those merge CAPZ can target the new release and have no more references to k8s.gcr.io

[the remark this answers is not in the document]
Comment (mboersma) [reply templates/addons/metrics-server/metrics-server.yaml:141]: This should be possible now that metrics-server [v0.6.3](https://github.com/kubernetes-sigs/metrics-server/releases/tag/v0.6.3) is released. See also comments in #3035.

**Pull request closed without merging**